src/lambda/handler-runner/docker-runner/DockerContainer.js

```diff
@@ -41,52 +45,78 @@ export default class DockerContainer {
 
     // TODO: support layer
     // https://github.com/serverless/serverless/blob/v1.57.0/lib/plugins/aws/invokeLocal/index.js#L291-L293
-    const dockerArgs = [
-      '-v',
-      `${codeDir}:/var/task:ro,delegated`,
-      '-p',
-      `${port}:9001`,
-      '-e',
-      'DOCKER_LAMBDA_STAY_OPEN=1', // API mode
-    ]
 
+    const envs = ['DOCKER_LAMBDA_STAY_OPEN=1']
     entries(this.#env).forEach(([key, value]) => {
-      dockerArgs.push('-e', `${key}=${value}`)
+      envs.push(`${key}=${value}`)
     })
 
+    const dockerArgs = {
+      Image: this.#imageNameTag,
+      Env: envs,
+      ExposedPorts: {
+        '9001/tcp': { HostPort: `${port}` },
+      },
+      HostConfig: {
+        PortBindings: { '9001/tcp': [{ HostPort: `${port}` }] },
+        Binds: [`${codeDir}:/var/task:ro,delegated`],
+      },
+      AttachStdin: false,
+      AttachStdout: true,
+      AttachStderr: true,
+    }
+
     if (platform() === 'linux') {
       // Add `host.docker.internal` DNS name to access host from inside the container
       // https://github.com/docker/for-linux/issues/264
       const gatewayIp = await this._getBridgeGatewayIp()
-      dockerArgs.push('--add-host', `host.docker.internal:${gatewayIp}`)
+      dockerArgs.HostConfig.ExtraHosts = [`host.docker.internal:${gatewayIp}`]
     }
 
-    const { stdout: containerId } = await execa('docker', [
-      'create',
-      ...dockerArgs,
-      this.#imageNameTag,
-      this.#handler,
-    ])
-
-    const dockerStart = execa('docker', ['start', '-a', containerId], {
-      all: true,
-    })
-
-    await new Promise((resolve, reject) => {
-      dockerStart.all.on('data', (data) => {
-        const str = data.toString()
-        console.log(str)
-        if (str.includes('Lambda API listening on port')) {
-          resolve()
-        }
+    const container = await this.#dockerClient.createContainer(dockerArgs)
+
+    const containerLogs = (runningContainer) => {
+      return new Promise((resolve, reject) => {
+        // create a single stream for stdin and stdout
+        const logStream = new stream.PassThrough()
+
+        logStream.on('data', (chunk) => {
+          if (chunk.toString('utf8').includes('Lambda API listening on port')) {
+            resolve()
+          }
+        })
```

Comment:
Please output stdout and stderr to log (https://github.com/dherault/serverless-offline/issues/768#issuecomment-580897730).